lib/email_domain_mx_validator.rb

```diff
@@ -0,0 +1,42 @@
+require 'resolv'
+
+module EmailDomainMxValidator
+  class Base
+    def check(domain)
+      raise 'Must implement and return a boolean'
+    end
+  end
+
+  class DnsStrategy < Base
+    def check(domain)
+      return false if domain.nil?
+
+      mx_records = Resolv::DNS.open do |dns|
+        dns.getresources(domain, Resolv::DNS::Resource::IN::MX)
+      end
+
+      return mx_records.size > 0
```

Comment:
That's right. It seems that VCR isn't even detecting it as a network request. So I left it untested because because I'd be testing implementation details.

Reply:
I created some tests that use the **real** DnsStrategy and I added a comment.
```
# makes a real DNS/HTTP request
EmailDomainMxValidator.strategy = EmailDomainMxValidator::DnsStrategy.new
```

Reply:
thank you